### ISSUE TITLE
Server watch functionality broken

### ISSUE THREAD

**Issue by joemckie** [joemckie/viva#1]
Currently the server inside Vagrant does not respond to changes made on the host machine, resulting in the server not being correctly reloaded. This is being looked into at https://github.com/Unitech/PM2/issues/1062. For the time being, please use the following temporary fix:

```
npm install pm2 -g
pm2 start processes.json
```

You can then access the server at `http://localhost:3000/`.

**Comment by CodingAspect** [comment]
I'm experiencing this issue as well, I haven't found any temp fix yet.
I'm using nodemon until it get fixed.

**Comment by joemckie** [comment]
Does nodemon work using the vagrant server or are you running it on the host machine? I tried that, but it didn't play nice with the Chef provisioning as it doesn't run in the background.

Thinking about it some more, the watch functionality mightn't be a major issue. It's well documented that the event has some latency on vagrant (usually a few seconds, but can be up to over a minute), so whilst developing the server should probably be ran on the host machine for maximum efficiency.

**Comment by CodingAspect** [comment]
Yes, nodemon works great using a vagrant server (over directory-sync).
I'm on OSX Yeosmite 10.10.3
pm2 0.14.3
vagrant 1.7.2
nodejs 12.0.7

I think it is a major issue because you don't expect a production server to use the 'watch' feature. It's only for development env.

**Comment by joemckie** [comment]
I see your point. I just tried nodemon using `vagrant ssh` and it works perfectly, so it seems like the latency varies wildly depending on what you're using to run the server. I'm looking into a solution to provisioning the server now as I'd ideally like everything to be set up on the `vagrant up` command without having to ssh into the machine post-install.

Will update as soon as I have results.
